beacon_node/execution_layer/src/test_utils/mock_builder.rs

```diff
@@ -58,119 +37,128 @@ pub enum Operation {
-            Operation::BlockNumber(block_number) => *bid.block_number_mut() = block_number as u64,
-            Operation::Timestamp(timestamp) => *bid.timestamp_mut() = timestamp as u64,
-            Operation::WithdrawalsRoot(root) => *bid.withdrawals_root_mut()? = to_ssz_rs(&root)?,
+            Operation::FeeRecipient(fee_recipient) => bid.set_fee_recipient(fee_recipient),
+            Operation::GasLimit(gas_limit) => bid.set_gas_limit(gas_limit as u64),
+            Operation::Value(value) => bid.set_value(value),
+            Operation::ParentHash(parent_hash) => bid.set_parent_hash(parent_hash),
+            Operation::PrevRandao(prev_randao) => bid.set_prev_randao(prev_randao),
+            Operation::BlockNumber(block_number) => bid.set_block_number(block_number as u64),
+            Operation::Timestamp(timestamp) => bid.set_timestamp(timestamp as u64),
+            Operation::WithdrawalsRoot(root) => bid.set_withdrawals_root(root),
         }
-        Ok(())
     }
 }
 
+#[derive(Debug)]
+struct Custom(String);
+
+impl warp::reject::Reject for Custom {}
+
 // contains functions we need for BuilderBids.. not sure what to call this
-pub trait BidStuff {
-    fn fee_recipient_mut(&mut self) -> &mut ExecutionAddress;
-    fn gas_limit_mut(&mut self) -> &mut u64;
-    fn value_mut(&mut self) -> &mut U256;
-    fn parent_hash_mut(&mut self) -> &mut Hash32;
-    fn prev_randao_mut(&mut self) -> &mut Hash32;
-    fn block_number_mut(&mut self) -> &mut u64;
-    fn timestamp_mut(&mut self) -> &mut u64;
-    fn withdrawals_root_mut(&mut self) -> Result<&mut Root, MevError>;
-
-    fn sign_builder_message(
-        &mut self,
-        signing_key: &SecretKey,
-        context: &Context,
-    ) -> Result<BlsSignature, Error>;
-
-    fn to_signed_bid(self, signature: BlsSignature) -> SignedBuilderBid;
+pub trait BidStuff<E: EthSpec> {
+    fn set_fee_recipient(&mut self, fee_recipient_address: Address);
+    fn set_gas_limit(&mut self, gas_limit: u64);
+    fn set_value(&mut self, value: Uint256);
+    fn set_parent_hash(&mut self, parent_hash: Hash256);
+    fn set_prev_randao(&mut self, randao: Hash256);
+    fn set_block_number(&mut self, block_number: u64);
+    fn set_timestamp(&mut self, timestamp: u64);
+    fn set_withdrawals_root(&mut self, withdrawals_root: Hash256);
+
+    fn sign_builder_message(&mut self, sk: &SecretKey, spec: &ChainSpec) -> Signature;
+
+    fn to_signed_bid(self, signature: Signature) -> SignedBuilderBid<E, BlindedPayload<E>>;
 }
 
-impl BidStuff for BuilderBid {
-    fn fee_recipient_mut(&mut self) -> &mut ExecutionAddress {
-        match self {
-            Self::Bellatrix(bid) => &mut bid.header.fee_recipient,
-            Self::Capella(bid) => &mut bid.header.fee_recipient,
+impl<E: EthSpec> BidStuff<E> for BuilderBid<E, BlindedPayload<E>> {
+    fn set_fee_recipient(&mut self, fee_recipient: Address) {
+        match self.header.to_mut() {
+            BlindedPayloadRefMut::Merge(payload) => {
```

Comment:
Wanted to rewrite this with a mapping macro but realised it isn't possible. Opened an issue on superstruct: https://github.com/sigp/superstruct/issues/31